app/src/main/java/co/nimblehq/ui/screens/home/HomeViewModel.kt

```diff
@@ -1,68 +1,83 @@
-package co.nimblehq.ui.main
+package co.nimblehq.ui.screens.home
 
 import androidx.hilt.lifecycle.ViewModelInject
 import co.nimblehq.common.transformers.Transformers
 import co.nimblehq.data.service.response.ExampleResponse
 import co.nimblehq.domain.repository.ApiRepository
 import co.nimblehq.domain.schedulers.BaseSchedulerProvider
-import co.nimblehq.lib.IsLoading
 import co.nimblehq.ui.base.BaseViewModel
-import co.nimblehq.ui.main.data.Data
+import co.nimblehq.ui.base.NavigationEvent
+import co.nimblehq.ui.screens.second.SecondBundle
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
 
-class MainViewModel @ViewModelInject constructor(
+interface Input {
+
+    fun refresh()
+
+    fun next()
+}
+
+class HomeViewModel @ViewModelInject constructor(
     private val repository: ApiRepository,
     private val schedulers: BaseSchedulerProvider
-) : BaseViewModel(), Inputs, Outputs {
-
-    private val refresh = PublishSubject.create<Unit>()
-    private val next = PublishSubject.create<Unit>()
-    private val gotoNext = PublishSubject.create<Data>()
+) : BaseViewModel(), Input {
 
-    private val data = BehaviorSubject.create<Data>()
-    private val isLoading = BehaviorSubject.create<IsLoading>()
+    private val _refresh = PublishSubject.create<Unit>()
+    private val _next = PublishSubject.create<Unit>()
 
-    val inputs: Inputs = this
-    val outputs: Outputs = this
+    private val _data = BehaviorSubject.create<Data>()
 
     init {
         fetchApi()
             .map { fromResponse(it) }
             .observeOn(schedulers.main())
             .subscribe({
-                data.onNext(it)
-                isLoading.onNext(false)
+                _data.onNext(it)
+                _showLoading.onNext(false)
             }, {
                 TODO("Handle Error  ¯\\_(ツ)_/¯ ")
             })
             .addToDisposables()
 
-        refresh
+        _refresh
             .flatMap<ExampleResponse> { fetchApi() }
             .map { fromResponse(it) }
             .observeOn(schedulers.main())
             .subscribe({
-                data.onNext(it)
-                isLoading.onNext(false)
+                _data.onNext(it)
+                _showLoading.onNext(false)
             }, {
                 TODO("Handle Error  ¯\\_(ツ)_/¯ ")
             })
             .addToDisposables()
 
-        data
-            .compose<Data>(Transformers.takeWhen(next))
+        _data
+            .compose<Data>(Transformers.takeWhen(_next))
             .subscribeOn(schedulers.io())
-            .subscribe(gotoNext::onNext)
+            .subscribe { _navigator.onNext(NavigationEvent.Second(SecondBundle(it))) }
```

Comment:
Could we update to `map` the navigation event before `subscribe`

Reply:
fixed